3ds2/src/main/java/com/adyen/checkout/adyen3ds2/Adyen3DS2ViewProvider.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 23/9/2022.
+ */
+
+package com.adyen.checkout.adyen3ds2
+
+import android.content.Context
+import android.util.AttributeSet
+import com.adyen.checkout.components.ui.ComponentViewNew
+import com.adyen.checkout.components.ui.PaymentInProgressView
+import com.adyen.checkout.components.ui.ViewProvider
+import com.adyen.checkout.components.ui.view.ComponentViewType
+
+object Adyen3DS2ViewProvider : ViewProvider {
+
+    override fun getView(
+        viewType: ComponentViewType,
+        context: Context,
+        attrs: AttributeSet?,
+        defStyleAttr: Int
+    ): ComponentViewNew = when (viewType) {
+        Adyen3DS2ComponentViewType -> PaymentInProgressView(context, attrs, defStyleAttr)
+        else -> throw IllegalStateException("Unsupported view type")
+    }
+}
+
+object Adyen3DS2ComponentViewType : ComponentViewType
```

Comment:
Are there any advantages of having a common `ViewType`/`ViewProvider` for all components who need it, in the same module as `PaymentInProgressView`?

Reply:
Hmm I have been thinking about that. In the end I decided to go with a specific type per component, because it gives more fine grained control and makes it easier to change the view (if that's ever needed)